DataLink.tex

```diff
@@ -888,26 +887,50 @@ \subsection{Service Resources}
-service is still operating as expected. The PARAM's description should
+service is still operating as expected. The PARAM description should
 give an indication of what the call will result in. End-user clients
-might indicate exampleURLs to the user after service failures.
-
-A GROUP with \attval{name}{inputParams} contains PARAM elements describing
-how to invoke the service. For services where the parameter values
-come from columns in the results resource, we use the ref attribute of
-the PARAM to indicate the FIELD (column) with the values. Other PARAM
-elements (without a ref attribute) are also allowed; these would describe
-additional service parameters, the type of value that must be specified,
-the meaning (UCD) of the value they apply to, etc.
-
+might indicate exampleURLs to the user after unexpected service failures.
+
+\subsection{Input PARAMs}
+
+A service descriptor must contain a GROUP element with \attval{name}{inputParams} 
+to describe user-specified input parameters of the service. There are three types of 
+input params: params with a fixed value, params where the values come from the 
+``results'', and params where the value is variable and chosen/specified by the user.
+
+For params with a fixed value (e.g. \attval{fly}{true}), the client {\bf must}
+treat it as a required parameter and include it in the service invocation; this allows
+a service implementor to include constant params explicitly (and describe them via a 
+DESCRIPTION element) rather than just include them in the ``accessURL'' without the
+possibility to explain them.
+
+For services where the parameter value(s) come from the ``results'' resource, the value 
+attribute is empty (\attval{value}{}) and the PARAM includes a ref attribute to indicate 
+the FIELD (column) that contains the values. For example, a TAP query result may contain 
+identifiers that can be used to invoke the {links} service; the FIELD with the identifiers 
+must have an XML ID attribute (e.g. \attval{ID}{abc}) and the input PARAM would include 
+the attribute \attval{ref}{abc}). When this mechanism is used, the client {\bf must}
+treat it as a required parameter and the parameter and value {\bf must} be included in 
+the service invocation.
+
+For user-specified input PARAMs the value attribute is normally empty (\attval{value}{}) 
+and the user supplies the value(s). The PARAM specifies the type of value required via 
+the datatype, arraysize, and xtype attributes; this may be augmented further by the ucd 
+and units attributes and a child DESCRIPTION element. In addition, the PARAM element allows 
+for additional child elements (such as MIN, MAX, or OPTIONS) to convey sensible constraints 
+on possible values. In general, services may have parameters of this type that are optional
+or required and this distinction is not currently described. Clients should assume that these
+user-specified parameters are optional, but that specifying at least one of them is necessary 
```

Comment:
The "at least one user-specified parameter" seems a bit arbitrary - may apply in some circumstances but not others.  I'd prefer to see advice along the lines of encouraging service descriptors to document optional/required constraints in DESCRIPTION elements than specifying a rule that might run counter to what some services need to do.